### PR TITLE
feat: upgrade brepkit-wasm 2.26.0 → 2.33.0

### PR DESCRIPTION
**Pull request by andymai**
## Summary
- Bump brepkit-wasm from 2.26.0 to 2.33.0 (12 upstream releases: new offset engine, sketch arcs, massive boolean GFA hardening)
- Add `scripts/sync-brepkit-types.ts` that auto-generates `brepkitWasmTypes.ts` from upstream `.d.ts` — eliminates type drift between releases
- Wire `offsetSolidV2` with feature detection (prefers V2 intersection-based offset engine, falls back to V1)
- Wire `sketchAddArc` + 6 new arc constraints (`tangentLineArc`, `tangentArcArc`, `pointOnArc`, `equalRadiusArcArc`, `arcLength`, `concentricArcArc`)
- Fix `ConstraintSketchCapability.sketchDof` return type: `number` → `string` (JSON) to match actual WASM behavior
- Remove stale version guards for `makeTangentArc3d` and `meshEdgesAll` (guaranteed in 2.33.0)

## Test plan
- [x] Full OCCT test suite: 150 files, 2525 tests passed
- [x] Full brepkit test suite: 150 files, 2525 tests passed
- [x] New tests: sketch arc entity/constraints, boolean edge cases (coplanar, identical-solid, NURBS)
- [x] `npm run validate` passes (typecheck + lint + boundaries + format)
- [x] `npm run knip` clean
- [x] `npm run check:patterns` clean (baseline updated)

### CODE REVIEW

File: tests/brepkitBooleanEdgeCases.test.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  it('cut identical solids produces error or empty', () => {
+    const a = box(10, 10, 10);
+    const b = box(10, 10, 10);
+    // Cutting identical shapes should produce an error (empty result) or throw
+    try {
+      const result = cut(a, b);
+      // If it doesn't throw, the result should be an error
+      expect(result.isOk).toBe(false);
+    } catch {
+      // Expected — brepkit throws for identical solid cut
```
[the remark this answers is not in the document]
Comment (andymai) [reply tests/brepkitBooleanEdgeCases.test.ts:22]: Fixed in 5c34b1e — `result.isOk` → `result.ok`.

File: src/kernel/brepkit/modifierOps.ts [hunk 2]
```diff
@@ -258,8 +258,12 @@ export function offset(
...
     return solidHandle(id);
   }
-  const id = bk.offsetSolid(unwrapSolidOrThrow(shape, 'offset'), distance);
-  return solidHandle(id);
+  const solidId = unwrapSolidOrThrow(shape, 'offset');
+  // Prefer V2 offset engine (intersection-based joints) when available
+  if (typeof bk.offsetSolidV2 === 'function') {
+    return solidHandle(bk.offsetSolidV2(solidId, distance));
+  }
+  return solidHandle(bk.offsetSolid(solidId, distance));
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/kernel/brepkit/modifierOps.ts]: Agreed — removed the guard entirely. Since we pin to 2.33.0 and `offsetSolidV2` is required in the interface, the feature detection was dead code. Now calls `bk.offsetSolidV2()` directly. Fixed in 5c34b1e.

File: scripts/sync-brepkit-types.ts [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+    methods: [
+      'makeBox', 'makeCylinder', 'makeSphere', 'makeCone', 'makeTorus',
+      'makeEllipsoid', 'makeRectangle', 'makePolygon', 'makeCircle',
+      'makeCircleFace',
+    ],
+  },
+  {
+    label: 'Shape construction',
+    methods: [
+      'makeVertex', 'makeLineEdge', 'makeCircleArc3d', 'makeTangentArc3d',
```
[the remark this answers is not in the document]
Comment (andymai) [reply scripts/sync-brepkit-types.ts:191]: Added a comment noting the limitation. wasm-bindgen output only produces flat primitive/typed-array params so this isn't a practical risk — but the note ensures future maintainers know to upgrade the parser if upstream changes. Fixed in 5c34b1e.
